### PR TITLE
Add Some Doxygen

### PR DESCRIPTION
**Pull request by Theundyingcode**
Started adding some deoygen comments to iexamine functions because it was annoying me that the only ones were for the method stubs I had added in a branch.

Is there a consensus on if the comments are going in the header or source file? After ~15 min of google browsing, I believe that source comments would be better for cataclysm.

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -72,12 +72,18 @@ static const trait_id trait_THRESH_MYCUS( "THRESH_MYCUS" );
...
+ * Nothing player can interact with here.
+ */
 void iexamine::none(player &p, const tripoint &examp)
 {
     (void)p; //unused
     add_msg(_("That is a %s."), g->m.name(examp).c_str());
 }
 
+/**
+ * Pick an apropriate item and apply dimond coating if possible.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "dimond" -> "diamond"

File: src/iexamine.cpp [hunk 2]
```diff
@@ -590,6 +610,9 @@ void iexamine::vending(player &p, const tripoint &examp)
     }
 }
 
+/**
+ * If there's water, allow it's usage but add chance of poison.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "it's" -> "its".

File: src/iexamine.cpp [hunk 3]
```diff
@@ -714,6 +749,9 @@ void iexamine::rubble(player &p, const tripoint &examp)
     p.invoke_item( *it, "DIG", examp );
 }
 
+/**
+ * Try to pry create open with crowbar.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "create" -> "crate".

File: src/iexamine.cpp [hunk 4]
```diff
@@ -744,6 +782,9 @@ void iexamine::crate(player &p, const tripoint &examp)
 }
 
 
+/**
+ * Prompt climbing over fence. Calculates move cost, applies it to player and, moves them.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: Excess comma after "and".

File: src/iexamine.cpp [hunk 5]
```diff
@@ -1057,6 +1125,13 @@ void iexamine::slot_machine( player &p, const tripoint& )
     }
 }
 
+/**
+ * Attempt to crack safe through audio-feedback manual lock manipulation.
+ * 
+ * Try to unlock the safe by moving the dial and listening for the mechanism to "click into place."
+ * Time per attempt affected by perception and mechanics. 30 minutes per atempt minimum.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "atempt" -> "attempt".

File: src/iexamine.cpp [hunk 6]
```diff
@@ -1162,6 +1243,9 @@ void iexamine::gunsafe_el(player &p, const tripoint &examp)
     }
 }
 
+/**
+ * Checks that player is outside and has crowbar then, calls iuse.crowbar.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: Excess comma after "then".

File: src/iexamine.cpp [hunk 7]
```diff
@@ -1210,6 +1293,9 @@ void iexamine::bulletin_board(player &, const tripoint &examp)
     }
 }
 
+/**
+ * Display popup with refrence to "The Enigma of Amigara Fault."
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "refrence" -> "reference".

File: src/iexamine.cpp [hunk 8]
```diff
@@ -1248,6 +1337,9 @@ void iexamine::pedestal_wyrm(player &p, const tripoint &examp)
     g->add_event(EVENT_SPAWN_WYRMS, int(calendar::turn) + rng(5, 10));
 }
 
+/**
+ * Put petrified eye on pedistal causing into to sink into ground open temple.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "pedistal" -> "pedestal".
"into" -> "it".

File: src/iexamine.cpp [hunk 9]
```diff
@@ -1268,6 +1360,9 @@ void iexamine::pedestal_temple(player &p, const tripoint &examp)
 large semi-spherical indentation at the top."));
 }
 
+/**
+ * Unlock/open door or atempt to peek through peephole.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "atempt" -> "attempt".

File: src/iexamine.cpp [hunk 10]
```diff
@@ -1361,6 +1456,9 @@ void iexamine::fswitch(player &p, const tripoint &examp)
     g->add_event(EVENT_TEMPLE_SPAWN, calendar::turn + 3);
 }
 
+/**
+ * If it's winter: show msg and, return true. Otherwise return false
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: Excess comma after "and".

File: src/iexamine.cpp [hunk 11]
```diff
@@ -1377,12 +1475,18 @@ bool dead_plant( bool flower, player &p, const tripoint &examp )
     return false;
 }
 
+/**
+ * Helper method to see if player has traits, hunger and mothwear for drinking nectar.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "mothwear" -> "mouthwear".

File: src/iexamine.cpp [hunk 12]
```diff
@@ -1395,6 +1499,12 @@ bool drink_nectar( player &p )
     return false;
 }
 
+/**
+ * Promt Pick (or drink nectar if able) poppy bud. Not safe for player.
+ * 
+ * Dinking causes: -25 hunger, +20 fatigue, pkill2-70 effect and, 1 in 20 pkiller-1 addiction.
+ * Picking w/ env_resist < 5 causes 1 in 3  sleep for 12 min and, 4 dmg to each leg
+ */
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "Promt" -> "Prompt".
"Dinking" -> "Drinking".
Excess commas after "and".

File: src/iexamine.cpp [hunk 13]
```diff
@@ -1554,6 +1670,9 @@ void iexamine::harvest_ter( player &p, const tripoint &examp )
     }
 }
 
+/**
+ * Only harvest a plant once per season.  Display mesage and call iexamine::none.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "mesage" -> "message".

File: src/iexamine.cpp [hunk 14]
```diff
@@ -1624,6 +1751,9 @@ void iexamine::egg_sackws( player &p, const tripoint &examp )
     egg_sack_generic( p, examp, mon_spider_web_s );
 }
 
+/**
+ * Remove furnature. Add spore effect.
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: "furnature" -> "furniture".

**Comment by Theundyingcode** [comment]
looks like I need an IDE with spell check...

**Comment by angoddu** [comment]
>> looks like I need an IDE with spell check...

It's very annoying that it isn't available by default in mode IDEs

**Comment by Theundyingcode** [comment]
@angoddu I agree. Especially since I just had to reset my computer and am taking the opportunity to shop for a good C++ IDE. (I've been using language specific ones or notepad++)

**Pull request merged**